adding obo permission to docs (#10332)
opensearch-project/documentation-website: _security/access-control/authentication-tokens.md

For more information, see the [related discussion](https://github.com/opensearch-project/security/issues/2891).

### Permissions

> In order to execute creation of OBO token the user needs to be assigned `security:obo/create` permission.

Reviewer: ```suggestion
To create an OBO token, you must have the `security:obo/create` permission.
```
